setup_environment.sh

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+#
+# Sets the environment needed for Hyperstyle to work.
+
+# ----------------------------------------------------------------------------------------------------------------------
+
+#######################################
+# Checks whether the linter needs to be installed.
+#
+# If the linter folder is empty, the linter must be installed,
+# otherwise, the linter must be reinstalled at the user's request.
+#
+# If the folder does not exist, it will be created.
+# If it is necessary to reinstall the linter, all files in the passed folder will be deleted.
+#
+# Arguments:
+#   Folder with the linter files, a path.
+# Returns:
+#   0 if the linter needs to be installed, 1 otherwise.
+#######################################
+function need_to_install_linter() {
+  if [[ -d $1 ]] && [[ -n $(ls -A "$1") ]]; then
+    read -p "The folder is not empty. Do you want to reinstall the linter? (Y/n): " -r
+    if [[ ! $REPLY =~ ^[Yy]$ ]]; then
+      return 1 # 1=false
+    fi
+    rm -rf "$1"
+  fi
+
+  mkdir -p "$1"
+  return 0 # 0=true
+}
+
+#######################################
+# Checks the return code.
+#
+# If the return code is non-zero, the error message is shown and the script
+# exits with exit status 1, otherwise the success message is shown.
+#
+# Arguments:
+#   Return code, an integer.
+#   Success message, a string.
+#   Error message, a string.
+# Outputs:
+#   Writes messages to stdout.
+#######################################
+function check_return_code() {
+  if [[ $1 -ne 0 ]]; then
+    echo "ERROR: $3"
+    exit 1
+  fi
+  echo "$2"
+}
+
+# ----------------------------------------------------------------------------------------------------------------------
+
+echo "Checking variables..."
+
+: "${CHECKSTYLE_VERSION:?Variable is not defined}"
+: "${CHECKSTYLE_DIRECTORY:?Variable is not defined}"
+
+: "${DETEKT_VERSION:?Variable is not defined}"
+: "${DETEKT_DIRECTORY:?Variable is not defined}"
+
+: "${PMD_VERSION:?Variable is not defined}"
+: "${PMD_DIRECTORY:?Variable is not defined}"
+
+echo "The variables are defined."
+
+echo
+
+read -p "Do you want to install ESLint? (Y/n): " -r
+if [[ $REPLY =~ ^[Yy]$ ]]; then
+  echo "Installing ESLint..."
+  npm install eslint@7.5.0 -g && eslint --init
+  check_return_code $? "ESLint installed." "ESLint installation failed."
+else
+  echo "ESLint installation skipped."
+fi
+
+echo
+
+read -p "Do you want to install Python requirements? (Y/n): " -r
```

Comment:
Please, separate it into two parts - general requirements and test requirements. 
Do you want to install Python requirements _for the Hyperstyle project_? - what do you think?